pkg/apis/resource/types.go

```diff
@@ -37,90 +299,332 @@ type ResourceClaim struct {
 	// +optional
 	metav1.ObjectMeta
 
-	// Spec describes the desired attributes of a resource that then needs
-	// to be allocated. It can only be set once when creating the
-	// ResourceClaim.
+	// Spec describes what is being requested and how to configure it.
+	// The spec is immutable.
 	Spec ResourceClaimSpec
 
-	// Status describes whether the resource is available and with which
-	// attributes.
+	// Status describes whether the claim is ready to use and what has been allocated.
 	// +optional
 	Status ResourceClaimStatus
 }
 
 // ResourceClaimSpec defines how a resource is to be allocated.
 type ResourceClaimSpec struct {
-	// ResourceClassName references the driver and additional parameters
-	// via the name of a ResourceClass that was created as part of the
-	// driver deployment.
-	ResourceClassName string
+	// Requests represent individual requests for distinct resources which must all be satisfied.
+	// If empty, nothing needs to be allocated.
+	//
+	// +optional
+	// +listType=atomic
```

Comment:
should this be
```
// +listType=map
// +listMapKey=name
```

Reply:
No, because the entire field is owned by one manager. See https://kubernetes.slack.com/archives/C0EG7JC6T/p1717485534890529 for a longer discussion around this.